controlplane/kubeadm/internal/cluster.go

```diff
@@ -102,27 +102,30 @@ func (m *Management) GetWorkloadCluster(ctx context.Context, clusterKey client.O
 		return nil, &RemoteClusterConnectionError{Name: clusterKey.String(), Err: err}
 	}
 
-	etcdCASecret := &corev1.Secret{}
-	etcdCAObjectKey := ctrlclient.ObjectKey{
-		Namespace: clusterKey.Namespace,
-		Name:      fmt.Sprintf("%s-etcd", clusterKey.Name),
-	}
-	if err := m.Client.Get(ctx, etcdCAObjectKey, etcdCASecret); err != nil {
-		return nil, errors.Wrapf(err, "failed to get secret; etcd CA bundle %s/%s", etcdCAObjectKey.Namespace, etcdCAObjectKey.Name)
-	}
-	crtData, ok := etcdCASecret.Data[secret.TLSCrtDataName]
-	if !ok {
-		return nil, errors.Errorf("etcd tls crt does not exist for cluster %s/%s", clusterKey.Namespace, clusterKey.Name)
-	}
-	keyData, ok := etcdCASecret.Data[secret.TLSKeyDataName]
-	if !ok {
-		return nil, errors.Errorf("etcd tls key does not exist for cluster %s/%s", clusterKey.Namespace, clusterKey.Name)
-	}
-
-	clientCert, err := generateClientCert(crtData, keyData)
+	// Retrieves the etcd CA key Pair
+	crtData, keyData, err := m.getEtcdCAKeyPair(ctx, clusterKey)
 	if err != nil {
 		return nil, err
 	}
+
+	// If the CA key is defined, the cluster is using a managed etcd, and so we can generate a new
+	// etcd client certificate for the controllers.
+	// Otherwise the cluster is using an external etcd; in this case the only option to connect to etcd is to re-use
+	// the apiserver-etcd-client certificate.
+	// TODO: consider if we can detect if we are using external etcd in a more explicit way (e.g. looking at the config instead of deriving from the existing certificates)
```

Comment:
In order to do so it is required to change the GetWorkloadCluster signature, and being this an invasive change I preferred to keep this out of scope of this PR